spec.md

```diff
@@ -190,18 +190,28 @@ within the same JSON object.
 ### source
 
 - Type: `URI-reference`
-- Description: This describes the event producer. Often this will include
-  information such as the type of the event source, the organization publishing
-  the event, the process that produced the event, and some unique identifiers.
-  The exact syntax and semantics behind the data encoded in the URI is event
-  producer defined.
+- Description: Identifies the event producer. Often this will include
+  information such as the type of the event source, the organization
+  publishing the event, the process that produced the event, and some unique
+  identifiers. The exact syntax and semantics behind the data encoded in the URI
+  is event producer defined.
+
+  An application MUST assign a distinct `source` to each distinct producer.
```

Comment:
At least this line and perhaps (but not necessarily) [250](https://github.com/cloudevents/spec/pull/391/files#diff-958e7270f96f5407d7d980f500805b1bR250) appear invalidated by #420.

Reply:
Fixed in next update. It reads a little strangely now because the rest of the current master spec is written as though source == producer, but it will make sense when #420 goes in.